### ISSUE TITLE
Icons not showing but not any errors

### ISSUE THREAD

**Issue by Millie290** [phocean/TopIcons-plus#135]
On fedora 31 the icons are not showing but it don't said that there is an error

**Comment by totaam** [comment]
Same here. The extension is active, everything looks OK, my application doesn't get any error from libappindicator either.

**Comment by robin-a-meade** [comment]
See if there are any error messages. In a terminal, run this command:

```
journalctl /usr/bin/gnome-shell -f -o cat
```

Then turn the extension off and back on and see if any errors are printed in your terminal. 

To turn the extension off/on you can use Gnome Tweaks app or the toggle at https://extensions.gnome.org/extension/1031/topicons/

**Comment by totaam** [comment]
This fixes it:
https://github.com/phocean/TopIcons-plus/commit/43f991d1533e8d4002bd25ae6afd8a1568c39b36#diff-06f1274e40de25abda72d812b1cce86a

**Comment by robin-a-meade** [comment]
@totaam  Good. 

The Fedora 31 package already has that commit. Thus, uninstall the one from https://extensions.gnome.org/extension/1031/topicons/ and then install the Fedora 31 package using:

```
sudo dnf install gnome-shell-extension-topicons-plus
```

Make sure `~/.local/share/gnome-shell/extensions/TopIcons@phocean.net` directory is not present, because that would take precedence over the system-wide Fedora package

**Comment by ericzolf** [comment]
The errors appeared for me also on Fedora 30 with Gnome 3.32 (gnome-shell-3.32.2-2.fc30.x86_64 and gnome-shell-extension-topicons-plus-22-5.fc30.20190929.ad2dd1a.noarch) so applying the patch manually from https://github.com/phocean/TopIcons-plus/issues/135#issuecomment-548639961 to `/usr/share/gnome-shell/extensions/TopIcons@phocean.net/extension.js` made it work again after a restart (despite the note that it would break 3.32 and below).

**Comment by proninyaroslav** [comment]
@robin-a-meade I installed the package (Fedora 31), but it doesn't work, it is tested on Telegram and qBittorrent. I encountered this problem after migrating from 30 to 31.

**Comment by robin-a-meade** [comment]
@proninyaroslav That's odd. I'll try those apps later. The apps I use it with are Dropbox, Guake,  Skype, Solaar, and VLC.

**Comment by robin-a-meade** [comment]
@ericzolf My experience is different. That commit is already present in the Fedora 30 package (version 22-5.20190929.ad2dd1a) and it causes the "Main.extensionManager is undefined" error when it is installed on Fedora 30 (which currently has Gnome 3.32.2).

**Comment by proninyaroslav** [comment]
Yes, VLC works. With Telegram and qBittorrent there is another oddity that I think is related to this, they have a weird window decoration: 
1. the title font is less clear than usual; 
2. no one-pixel separator between the window contents and the window title; 
3. if I hover over the "Close" button, there is no highlight animation, although the button works.

**Comment by proninyaroslav** [comment]
I think this is due to the decoration of non-GTK apps, since both of them are written in Qt.

**Comment by ericzolf** [comment]
@robin-a-meade you're right, sorry, I read the patch the wrong way around and _unapplied_ it to make top icon work again on F30. PR #136 is probably the proper solution.

**Comment by gloomytrousers** [comment]
Looks to be the same as #131.

**Comment by phocean** [comment]
I merged #136 so this should have been fixed.

I am closing it but feel free to reopen if the issue still arises.
